### PR TITLE
Language tests

### PR DESCRIPTION
**Pull request by Fatalll**

### CODE REVIEW

File: src/test/resources/ide/language/python/Example.py [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+    I <warning>has</warning> <warning>many error</warning> in this <warning>sentnce</warning>
+"""
+
+dictionary = dict()
+dictionary['kei 1'] = '<warning>this string have</warning> <warning>a</warning> error'
+dictionary["key 2"] = "<warning>the a</warning> double <warning>quoutes</warning> error"
+
+
+def area_by_shoelace(x, y):
```
Comment (TanVD) [inline src/test/resources/ide/language/python/Example.py]: Lets additionally check for:
* strings as arguments of functions
* templated strings
Also it is better to structure tests by format of string and the way it is used. See StringLiteral.kt for an example

Comment (Fatalll) [reply src/test/resources/ide/language/python/Example.py]: Split tests, add tests for string literal passing to function

File: src/test/resources/ide/language/js/Constructs.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+class Some<warning>Classs</warning> {
+
+    constructor(config) {
+        this.member_varibleleel = config;
+    }
+
+    some_<warning>metod</warning>() {
+        console.log(this.member_varibleleel);
+    }
+}
```
Comment (TanVD) [inline src/test/resources/ide/language/js/Constructs.js]: Setup at IDEA settings "Add newline at end of file" please

File: src/test/resources/ide/language/js/Constructs.js [hunk 3]
```diff
@@ -0,0 +1,13 @@
+
+var some_<warning>constnt</warning> = "1";
+
+class Some<warning>Classs</warning> {
+
+    constructor(config) {
+        this.member_varibleleel = config;
```
Comment (TanVD) [inline src/test/resources/ide/language/js/Constructs.js]: Add here //TODO tag :)

File: src/test/resources/ide/language/js/Constructs.js (hunk 2 above)
Comment (Fatalll) [reply src/test/resources/ide/language/js/Constructs.js]: Done

File: src/test/resources/ide/language/js/Constructs.js (hunk 3 above)
Comment (Fatalll) [reply src/test/resources/ide/language/js/Constructs.js]: Done

File: src/test/kotlin/tanvd/grazi/ide/language/XMLSupportTest.kt [hunk 4]
```diff
@@ -0,0 +1,14 @@
+package tanvd.grazi.ide.language
+
+import tanvd.grazi.GraziTestBase
+
+
+class XMLSupportTest : GraziTestBase(true) {
```
Comment (TanVD) [inline src/test/kotlin/tanvd/grazi/ide/language/XMLSupportTest.kt:6]: Does our support work for HTML? If yes, let's add it here too

File: src/test/resources/ide/language/js/Constructs.js [hunk 5]
```diff
@@ -0,0 +1,13 @@
+
+var some_<warning>constnt</warning> = "1";
+
+class Some<warning>Classs</warning> {
+
+    constructor(config) {
+        this.member_varibleleel = config; // TODO Check member names
```
Comment (TanVD) [inline src/test/resources/ide/language/js/Constructs.js]: By the way, for some reason with last plugin in ultimate this typo works

File: src/test/resources/ide/language/js/JSDoc.js [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+    /** methodOne description */
+    methodOne() {
+        console.log("methodOne");
+    }
+
+    //Another method. Receives a Fruit object parameter, public
+    /**
+     * methodTwo description
+     * @param  {Object} fruit      param <warning>descriproin</warning>
+     * @param  {String} fruit.nme  nme of the fruit
```
Comment (TanVD) [inline src/test/resources/ide/language/js/JSDoc.js]: `nme` is not highlighted? Probably need to debug

File: src/test/resources/ide/language/js/Constructs.js (hunk 5 above)
Comment (TanVD) [reply src/test/resources/ide/language/js/Constructs.js]: Probably need to debug :)

File: src/test/resources/ide/language/python/StringLiterals.py [hunk 7]
```diff
@@ -0,0 +1,17 @@
+oneTypo = "It is <warning>friend</warning> of human"
```
Comment (TanVD) [inline src/test/resources/ide/language/python/StringLiterals.py:1]: We need to add multilines here :)

File: src/test/resources/ide/language/python/DocString.py [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+
+This script allows the user to print to the console all columns in the
+spreadsheet. <warning>It is assumed that</warning> the first row of the spreadsheet is the
+location of the columns.
+
+"""
+
+
+def test(fil_loc, print_cols=False):
+    """Gets and prints the spreadsheet's header columns
```
Comment (TanVD) [inline src/test/resources/ide/language/python/DocString.py]: Cleanup unneeded strings here

File: src/test/resources/ide/language/rust/Constructs.rs [hunk 9]
```diff
@@ -0,0 +1,23 @@
+
+pub struct <warning>Peerson</warning> {
+    <warning>naame</warning>: String,
+}
+
+impl Peerson {
+
```
Comment (TanVD) [inline src/test/resources/ide/language/rust/Constructs.rs:8]: Cleanup here and everywhere unneded whitespaces

File: src/test/resources/ide/language/js/JSDoc.js (hunk 6 above)
Comment (Fatalll) [reply src/test/resources/ide/language/js/JSDoc.js]: Are you sure? nme is a variable name, so no highlighting...

File: src/test/resources/ide/language/python/StringLiterals.py (hunk 7 above)
Comment (Fatalll) [reply src/test/resources/ide/language/python/StringLiterals.py:1]: Do you mean triple quotes? They are testing in DocString.

File: src/test/resources/ide/language/xml/Example.xml [hunk 10]
```diff
@@ -0,0 +1,24 @@
+<?xml version="1.0" encoding="UTF-8"?>
+
+<shiporder orderid="889923"
```
Comment (TanVD) [inline src/test/resources/ide/language/xml/Example.xml]: Remove unneeded attributes

File: src/test/resources/ide/language/xml/Example.xsd [hunk 11]
```diff
@@ -0,0 +1,51 @@
+<?xml version="1.0" encoding="UTF-8" ?>
+<xs:schema xmlns:xs="http://www.w3.org/2001/XMLSchema">
```
Comment (TanVD) [inline src/test/resources/ide/language/xml/Example.xsd:2]: Cleanup unneeded lines :)

File: src/test/kotlin/tanvd/grazi/ide/language/XMLSupportTest.kt (hunk 4 above)
Comment (Fatalll) [reply src/test/kotlin/tanvd/grazi/ide/language/XMLSupportTest.kt:6]: In the same class, or create another test?

Comment (TanVD) [reply src/test/kotlin/tanvd/grazi/ide/language/XMLSupportTest.kt:6]: In the same class :)

File: src/test/resources/ide/language/js/JSDoc.js (hunk 6 above)
Comment (TanVD) [reply src/test/resources/ide/language/js/JSDoc.js]: I mean on the right `nme of the fruit`

File: src/test/resources/ide/language/python/StringLiterals.py (hunk 7 above)
Comment (TanVD) [reply src/test/resources/ide/language/python/StringLiterals.py:1]: Well, ok)

File: src/test/resources/ide/language/js/JSDoc.js (hunk 6 above)
Comment (Fatalll) [reply src/test/resources/ide/language/js/JSDoc.js]: Default BaseSplitter ignore all words less than 4 characters

Comment (TanVD) [reply src/test/resources/ide/language/js/JSDoc.js]: Yep, solved :)

File: src/test/resources/ide/language/js/Constructs.js (hunk 5 above)
Comment (Fatalll) [reply src/test/resources/ide/language/js/Constructs.js]: Also doesn't work in Idea, so it's a feature :)

File: src/test/kotlin/tanvd/grazi/ide/language/XMLSupportTest.kt (hunk 4 above)
Comment (Fatalll) [reply src/test/kotlin/tanvd/grazi/ide/language/XMLSupportTest.kt:6]: Done

File: src/test/resources/ide/language/python/DocString.py (hunk 8 above)
Comment (Fatalll) [reply src/test/resources/ide/language/python/DocString.py]: Done

File: src/test/resources/ide/language/rust/Constructs.rs (hunk 9 above)
Comment (Fatalll) [reply src/test/resources/ide/language/rust/Constructs.rs:8]: Done

File: src/test/resources/ide/language/xml/Example.xml (hunk 10 above)
Comment (Fatalll) [reply src/test/resources/ide/language/xml/Example.xml]: Simplify schema

File: src/test/resources/ide/language/xml/Example.xsd (hunk 11 above)
Comment (Fatalll) [reply src/test/resources/ide/language/xml/Example.xsd:2]: done

**Pull request merged**